src/query/plan/rule_based_planner.cpp

```diff
@@ -516,14 +517,30 @@ bool HasBoundFilterSymbols(const std::unordered_set<Symbol> &bound_symbols, cons
 
 Expression *ExtractFilters(const std::unordered_set<Symbol> &bound_symbols, Filters &filters, AstStorage &storage) {
   Expression *filter_expr = nullptr;
+  std::vector<FilterInfo> and_joinable_filters{};
```

Comment:
Not sure I understand the logic, why you need inner loop which iterates over and_joinable_filters? Shouldn't it be enough that you iterate over all filters and you just skip those with Type::Pattern?

Reply:
Idea here is to push Type::Pattern filters do the end of loop, so that is why I am iterating always over all of them, and if it is pattern filter, then I insert element just before  that Pattern filter element. 

So if you have
Current = Generic


                                                        1            2           3           4
and_joinable_filters:                    Generic Generic Generic Pattern

I want to insert Generic before PAttern filter, so if you have multiple ANDS, if one of the generic fails, I want to stop execution and not evaluate pattern

Reply:
aha I think you can then use std::partition that should be simpler and faster :)